### PR TITLE
Enhance `ProcessingExporter` system to support redaction of content in telemetry traces

### PR DESCRIPTION
**Pull request by mpenn**
## Description

Closes https://github.com/NVIDIA/NeMo-Agent-Toolkit/issues/756

Agentic applications will somtimes interact with sensitive data sources, increasing the sensitivity of observability traces transmitted to external systems. This PR extends the `ProcessingExporter` system to address this concern. It features a `RedactionProcessor` and `SpanTaggingProcessor` allowing the application to conditionally redact content and tag Spans with privacy classifications to enable downstream trace access control.  Additionally, it increases the flexibility of processing pipeline configuration, allowing processors to be named, replaced by name/position/instance, and inserted anywhere in an existing pipeline.

An example configuration

```
# Uses the new mixin classes to add in additional configuration options
class OtelCollectorRedactionTelemetryExporter(HeaderRedactionConfigMixin,
                                              PrivacyTaggingConfigMixin,
                                              OtelCollectorTelemetryExporter,
                                              name="otelcollector_redaction"):
    """A telemetry exporter to transmit traces to externally hosted otel collector service with redaction."""

    resource_attributes: dict[str, str] = Field(default_factory=dict,
                                                description="The resource attributes to add to the span")


@register_telemetry_exporter(config_type=OtelCollectorRedactionTelemetryExporter)
async def otel_telemetry_redaction_exporter(config: OtelCollectorRedactionTelemetryExporter, builder: Builder):
    """Create an OpenTelemetry telemetry exporter."""

    from nat.plugins.opentelemetry.otel_span_exporter import get_opentelemetry_sdk_version
    from nat.plugins.opentelemetry.otlp_span_redaction_adapter_exporter import OTLPSpanHeaderRedactionAdapterExporter

    # Default resource attributes
    default_resource_attributes = {
        "telemetry.sdk.language": "python",
        "telemetry.sdk.name": "opentelemetry",
        "telemetry.sdk.version": get_opentelemetry_sdk_version(),
        "service.name": config.project,
    }

    # Merge defaults with config, giving precedence to config
    merged_resource_attributes = {**default_resource_attributes, **config.resource_attributes}

   # Function that will serve as the bass of conditional content redaction
   def my_redaction_callback(api_key: str) -> bool:
       return some_action_with_key(api_key)

    # Initialize a redaction exporter that uses header information as the basis of conditional redaction
    yield OTLPSpanHeaderRedactionAdapterExporter(endpoint=config.endpoint,
                                                                                   resource_attributes=merged_resource_attributes,
                                                                                   batch_size=config.batch_size,
                                                                                   flush_interval=config.flush_interval,
                                                                                   max_queue_size=config.max_queue_size,
                                                                                   drop_on_overflow=config.drop_on_overflow,
                                                                                   shutdown_timeout=config.shutdown_timeout,
                                                                                   redaction_attributes=config.redaction_attributes,
                                                                                   redaction_header=config.redaction_header,
                                                                                   redaction_callback=my_redaction_callback,
                                                                                   redaction_enabled=config.redaction_enabled,
                                                                                   force_redaction=config.force_redaction,
                                                                                   privacy_tag_key=config.tag_key,
                                                                                   privacy_tag_value=config.tag_value)
```

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/NeMo-Agent-Toolkit/blob/develop/docs/source/resources/contributing.md).
- We require that all contributors "sign-off" on their commits. This certifies that the contribution is your original work, or you have rights to submit it under the same license, or a compatible license.
  - Any contribution which contains commits that are not Signed-Off will not be accepted.
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

## Summary by CodeRabbit

- **New Features**
  - Header-based span redaction (configurable attributes, header key, callback, force-redact, redaction value).
  - Span privacy tagging (configurable key/value, environment-driven prefix).
  - OTLP exporter variant integrating header redaction and privacy tagging.

- **Improvements**
  - Processing pipeline: named processors, flexible insert/remove/positioning, pipeline locking/reset, optional dropping of None results.
  - New redaction/tagging configuration mixins and pluggable redaction framework.
  - OpenTelemetry exporters re-exported at package root.

- **Tests**
  - Extensive new and expanded tests covering redaction, tagging, pipeline behavior, exporters, batching, and edge cases.

- **Documentation**
  - Patronus exporter link updated.

### CODE REVIEW

File: docs/source/tutorials/create-a-new-workflow.md [hunk 1]
```diff
@@ -198,7 +198,7 @@ async def text_file_ingest_function(config: TextFileIngestFunctionConfig, builde
 
 ## Creating the Workflow Configuration
 
-Starting from the `custom_config.yml` file you created in the previous section, replace the two `webpage_query` tools with the new `text_file_ingest` tool. For the data source, you can use a collection of text files located in the `examples/documentation_guides/workflows/text_file_ingest/data` directory that describes [DOCA GPUNetIO](https://docs.nvidia.com/doca/sdk/doca+gpunetio/index.html).
+Starting from the `custom_config.yml` file you created in the previous section, replace the two `webpage_query` tools with the new `text_file_ingest` tool. For the data source, you can use a collection of text files located in the `examples/documentation_guides/workflows/text_file_ingest/data` directory that describes [DOCA GPUNetIO](https://docs.nvidia.com/doca/archive/3-0-0/doca+gpunetio/index.html).
```
Comment (willkill07) [inline docs/source/tutorials/create-a-new-workflow.md]: I think #755 already fixes this with a newer URL

File: src/nat/observability/mixin/tagging_config_mixin.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+from typing import Generic
+from typing import TypeVar
+
+from pydantic import BaseModel
+from pydantic import Field
+
+TagValueT = TypeVar("TagValueT")
+
+
+class PrivacyLevel(Enum):
```
Comment (willkill07) [inline src/nat/observability/mixin/tagging_config_mixin.py]: nit: include (str, Enum) as inheritance list

File: src/nat/observability/exporter/processing_exporter.py [hunk 3]
```diff
@@ -99,27 +127,232 @@ async def pipeline_callback(item):
...
+                raise ValueError(f"Processor '{processor}' not found in pipeline")
+            position = self._processor_names[processor]
+            processor_obj = self._processors[position]
+        elif isinstance(processor, int):
+            # Remove by position
+            if not (0 <= processor < len(self._processors)):
+                raise ValueError(f"Position {processor} is out of range [0, {len(self._processors) - 1}]")
+            position = processor
+            processor_obj = self._processors[position]
+        elif hasattr(processor, 'process'):  # Duck typing check for Processor
```
Comment (willkill07) [inline src/nat/observability/exporter/processing_exporter.py]: Does subclass check not work here from Processor?

**Review by willkill07 — APPROVED** [review]
Just minor nits. Always a joy to review your PRs.
